### PR TITLE
chore(rpc): move `EthApi` trait into own crate

### PR DESCRIPTION
**Pull request by emhane**
Moves `EthApi` trait, and building-block traits, into own crate. This makes it possible to impl `EthApiServer` for a generic type that impl the building-block traits.

This way, small parts of the `EthApiServer` impl, can be customised, by overriding the default impl of any of the building-block traits.

### CODE REVIEW

File: crates/node-core/src/args/gas_price_oracle.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 use crate::primitives::U256;
 use clap::Args;
-use reth_rpc::eth::gas_oracle::GasPriceOracleConfig;
+use reth_rpc_eth_api::GasPriceOracleConfig;
```
Comment (mattsse) [inline crates/node-core/src/args/gas_price_oracle.rs:3]: I want to avoid these changes, instead this should be re-exported from reth-rpc

**Review by mattsse — CHANGES_REQUESTED** [review]
supportive

I want to keep changes in other crates minimal and I think we can prevent this by doing re-exports from reth-rpc

File: crates/node-core/src/args/gas_price_oracle.rs (hunk 1 above)
Comment (emhane) [reply crates/node-core/src/args/gas_price_oracle.rs:3]: `eth` is the only `reth-rpc` module used in `reth-node-core` builder, hence why I went with updating the dep here

**Comment by emhane** [comment]
incremental progress committed in https://github.com/paradigmxyz/reth/pull/8887

**Pull request closed without merging**